### PR TITLE
DDT-884 re-introduced schema validation and applied marshalling rules…

### PR DESCRIPTION
**Pull request by pimg**
Resolves: DDT-884

### CODE REVIEW

File: src/main/java/org/dcsa/ebl/Application.java [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+  @Bean
+  public ObjectMapper objectMapper() {
+    return new ObjectMapper()
+        .enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+        .enable(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES)
+        .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+        .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
+        .setDateFormat(new SimpleDateFormat("yyyy-MM-dd"))
+        .findAndRegisterModules();
+  }
```
Comment (nt-as) [inline src/main/java/org/dcsa/ebl/Application.java:24]: Should this be the default for all projects and if so, should we move it to DCSA-Event-Core or DCSA-Core instead in the long run?

File: src/main/java/org/dcsa/ebl/controller/TransportDocumentController.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
       @PathVariable @Size(max = 20) String transportDocumentReference) {
     return transportDocumentService
         .findByTransportDocumentReference(transportDocumentReference)
+      .map(transportDocumentTO -> transportDocumentTO)
```
Comment (nt-as) [inline src/main/java/org/dcsa/ebl/controller/TransportDocumentController.java:32]: Maybe a `.cast(...)` instead (or `.<Type>findByTransportDocumentReference(transportDocumentReference)`)

File: src/main/java/org/dcsa/ebl/Application.java (hunk 1 above)
Comment (pimg) [reply src/main/java/org/dcsa/ebl/Application.java:24]: @nt-as  yes, I created a techdebt ticket for this: https://dcsa.atlassian.net/browse/DDT-909

File: src/main/java/org/dcsa/ebl/controller/TransportDocumentController.java (hunk 2 above)
Comment (pimg) [reply src/main/java/org/dcsa/ebl/controller/TransportDocumentController.java]: This was used during debugging only to properly set a breakpoint. is does not serve a real purpose, so is deleted

**Review by nt-as — APPROVED** [review]

**Pull request merged**